### ISSUE TITLE
Problems with Jaeger all-in-one datasource

### ISSUE THREAD

**Issue by renato-viasoft** [grafana/grafana#96676]
## Quick disclaimer
In a couple months now, i've been implementing a observability environment based on multi-tenancy for instrumenting the projects in my organization. While working on traces, some problems have appeared and i could'nt find much documentation or tutorials in the matter. For those reasons i am creating this issue. 

## Services used
- Grafana: v11.2.0
- Jaeger: jaegertracing/all-in-one 1.63.0 (latest)
  - Note: Jaeger is beeing suplied by data collected from a Open Telemetry Collector.

#### The traces are shown in the table visualization
![Image](https://github.com/user-attachments/assets/eb846fdd-3d99-42e6-a399-1ee3b39d0d10)
#### But not in the traces view
![Image](https://github.com/user-attachments/assets/7351b002-4664-4ff0-98aa-36b1cabecbcd)

#### When clicking in the traceId in the table view
![Image](https://github.com/user-attachments/assets/ac4639a1-bf01-42ac-a7e4-036ae322c8f9)
#### The user is redirected to the explore tab
![Image](https://github.com/user-attachments/assets/1e54a2f7-66ff-4a94-af27-16f5a8bebd61)
Which has restricted access.

#### When accessing the explore for traces
![Image](https://github.com/user-attachments/assets/e332e1a7-4408-46a5-94a7-6f54c0859e55)
#### Jaeger is not avaliable to access in the data source option
![Image](https://github.com/user-attachments/assets/f4b8e8ab-6d3e-48d6-a975-c02b9c139c3d)

#### When trying to view the node graph of services dependencies, it does not show
![Image](https://github.com/user-attachments/assets/285ee306-b0a3-4dbd-8ce1-83c54c5d9fab)
It is a strange occourence since i had run the Jaeger schedule manually to create that data since it seems all-in-one does not do that automatically. Unfortunally i cannot show that graph because i dont have access to that anymore, for that, please consider this a secondary problem.

### What did you expect to happen?

My main concern is the visualization of individual traces requiring admin role, it is a great security concern grant those privilleges to all users and it is something that i will not do. For that i think that there should be another way to visualize that information, i would suggest a visualization for individual traces and it would preferably be accessed when clicking on the traceId.

Other than that, the traces visualization not workins concerns me because it could mean that i am not showing important information or it could be the solution to my previous problem of viewing individual traces so if it is possible i would like a clarification on the reason that it is not working.

### Did this work before?

No, ever since i've started using grafana it did not work. I have searched for solutions but could not make it work. I am reporting this issue just now because my project is starting to be used so the problem has turned more critical than before.

### How do we reproduce it?

1. Run a tracing observability environment with Jaeger and Open Telemetry on Spring Boot Applications
2. On Grafana, add a Jaeger data source
3. Create a Visualization
4. Try to acceess a individual trace with a user that is not admin

### Is the bug inside a dashboard panel?

| Key | Value |
|--|--|
| Panel | table @ 11.2.0 |
| Grafana | Grafana v11.2.0 (2a88694fd3) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          },
          "color": {
            "mode": "thresholds"
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "showHeader": true,
        "cellHeight": "sm",
        "footer": {
          "show": false,
          "reducer": [
            "sum"
          ],
          "countRows": false,
          "fields": ""
        }
      },
      "pluginVersion": "11.2.0",
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "meta": {
                  "preferredVisualisationType": "table"
                },
                "fields": [
                  {
                    "name": "traceID",
                    "type": "string",
                    "config": {
                      "unit": "string",
                      "displayNameFromDS": "Trace ID",
                      "links": [
                        {
                          "title": "Trace: ${__value.raw}",
                          "url": "",
                          "internal": {
                            "datasourceUid": "ee1mwvbydzfuod",
                            "datasourceName": "jaeger",
                            "query": {
                              "query": "${__value.raw}"
                            }
                          }
                        }
                      ]
                    }
                  },
                  {
                    "name": "traceName",
                    "type": "string",
                    "config": {
                      "displayNameFromDS": "Trace name"
                    }
                  },
                  {
                    "name": "startTime",
                    "type": "time",
                    "config": {
                      "displayNameFromDS": "Start time"
                    }
                  },
                  {
                    "name": "duration",
                    "type": "number",
                    "config": {
                      "displayNameFromDS": "Duration",
                      "unit": "µs"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    "65ca6b16b59bd4a10f9336f98dc12ae9",
                    "90fb9406dd07ca043363cf1765919e52",
                    "8c3919dd71362cc4df79e82f178c87aa",
                    "ad5c7c0680ec409ee97f299fd5a75151",
                    "bf1ea720d9b1326580f9d77943a5bea7",
                    "11a2fb3d4be05cb994c412c4e6a061e8",
                    "d58fc0024b92c734649ed766e3cb9cf6",
                    "209d790caaebe52c269ec1c9f8f38c7d",
                    "5e062e5555b1e85b5a77943f326ac53f",
                    "363c513a2a0c96f5366cbf550c947f9a",
                    "77dfb4f65c7cfc4e763a9c1d07526fe0",
                    "2e93426312d7db9ed958adfa2ab5e372",
                    "2fb5b10aabea12096fb7a52e48ceb997",
                    "5992629f2b8a01ff0c3198202e2e0b84",
                    "73918f261a837c6f8832c689114cbfa9",
                    "7990c41f284dc7fe82d46ddbe1bd5312",
                    "055033f78cc7566f8a6dfa4ccdc5d7e8",
                    "a940b8bab2e98d41d74b070168887013",
                    "a0f20f8b9857f24448678b66a0e1c4a0",
                    "3aa2f9d01e7f292d7014a7b45b44ac8e",
                    "d5868bb11d7df03241ca971d773be62f",
                    "1b59fdac9c4dca90824ef10ba6fbc8c8",
                    "7adc5e031dfa2a27f71d5cc991f0903a",
                    "c43b6cba40e1c65569e1e94d28d78c74",
                    "ee6c165534a3850e31c9aac6410a3c61",
                    "c633dbc8fbb4e36ce0976bc6daa75416",
                    "7df742dce1a6673d86e4d05fb0b16c73",
                    "030b2c2834c5276359e5c8af7cddc049",
                    "326c9242909a69f7e456d472a04bc33e",
                    "2b0021fdb599af37c765f66266bc3c25",
                    "c62485af1797bbaf096e9fd5856887e4",
                    "931fc1348cc7d2444af87c8ef7372573",
                    "93e2d77b5197fd149afb1aadaa0727e9",
                    "a0aabab8fdeb2a658acaa065d1ade7e0",
                    "aafe211fe4639b3a8b58534defa77731",
                    "b7f2022eee895321f618f17ad1ab3d2d",
                    "cd58b7c71397302542f06759832e7fe8",
                    "c93e70978222f3d28abe4a744f364c20",
                    "38fb9c3ba25a6dee01503f91ef32547d",
                    "7d161db6709d43a8113627a2133a72b6",
                    "3ead9cbace75e054904718119bf904f1",
                    "edfc780fd8fef41ea737920107a9632a",
                    "a97b3a03a39adf883e49c44b7ae4c8cf",
                    "d3ec6b528a8821dc7e65c29f7cc84216",
                    "1eeb9c011062400d6bdd0990ed97a674",
                    "7f0f930929adc43e2993d031e675bee9",
                    "7bcf91e5eafddd56069e1920166bdad1",
                    "c10bf69b672fc41a79ae0043ab5f5856",
                    "b63d367a22001780356ddd26edfe6db2",
                    "cc3a3aea18ba79f0a065ad5b31e61bf2",
                    "607705fc91f1e2a91415f4a0a51c2365",
                    "8f07d589195fe18c8d0932d45b070278",
                    "d6f014c048d7f3b79eadfed0cef9a6f2",
                    "039890c5e41513dfd5dae34a138f2226",
                    "8ab79ac942be7e9bf91bb3921c98f008",
                    "94eb71bff17feda7b456d10992e538c5",
                    "b256cd50d479a92c29f6e131561e2423",
                    "ad31947aa5ce6dabf919f99df0571da6",
                    "7459bcb62298ed57120b6dc10afa0250",
                    "f2420cd07a00bc54c1aff9043800be60",
                    "3e636cda373c75c36f007be7c8e81c40",
                    "ac6dc19b666ea447b8867eff189fdcfc",
                    "b84834490870711c23e8b29e489ce8de",
                    "49ad13c27cc971d985b037d5553fe3f3",
                    "686c37cc8e3c4e42f55f06a2aa3f1364",
                    "c6539d9ee33080b484ef1f7e7bca23c1",
                    "d0ab8549be40f7a4fcd577f4373c1adf",
                    "92e6c2f763bed6bcef3a4e586b89789d",
                    "789726e03971e132aba2333cc3e672c0",
                    "89dc6a71f146f9c23a1a48eb2f8e2985",
                    "738f4d3aa847df5df62119136f929815",
                    "2b20d38079640b2e4188e72ef0e77c59",
                    "f2446a62ec0110191d7f5597b167ad97",
                    "325344e4ff79d19dad0c2b2221239ded",
                    "27226ce2fb4467e8d1e7c92ea68e1ca7",
                    "59e62deae99eb8b646bc942a9f28bedc",
                    "1c86e400f3648004a4727650bcfd05f1",
                    "117dfb39ff4e20e096c8e8d803851abe",
                    "b5c91c742a9339cd7ca427d88554a9b8",
                    "5a3c8854d848fea79bdfe9450998d33b",
                    "7e91bb4a0b19b440474899060dc9feff",
                    "d4e2328385c3affa389ef167822b3461",
                    "dd02c7bce5ea3e0e3d890980a68fe20f",
                    "34273c59b0c7280e725a3ada0f7b5903",
                    "06419c93bd1bcc71ac9a60f574531d12",
                    "f9cd63fc475d93b25718f832eea57140",
                    "926eed3a8a99586ec4bb0f7471643ecd",
                    "3134dee04551e11e61a7cd0b4bf87f23",
                    "2d0ee9390740c99d0260c0990bc0f247"
                  ],
                  [
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET",
                    "ratelimit: GET",
                    "ratelimit: GET",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: POST /**",
                    "ratelimit: POST",
                    "ratelimit: POST",
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: POST /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: HEAD /**",
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: HTTP",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: GET /**",
                    "ratelimit: POST",
                    "ratelimit: GET /**"
                  ],
                  [
                    1732020897087.932,
                    1732020035929.879,
                    1732020026529.671,
                    1732020018995.535,
                    1732019654366.667,
                    1732019653639.567,
                    1732019652717.153,
                    1732019651781.46,
                    1732019650828.125,
                    1732019650199.753,
                    1732019649561.8,
                    1732019648961.706,
                    1732019648292.271,
                    1732019647500.782,
                    1732019646729.503,
                    1732019645909.293,
                    1732019644983.595,
                    1732019644085.207,
                    1732019643294.493,
                    1732019642508.899,
                    1732019641701.191,
                    1732019640781.909,
                    1732019639954.086,
                    1732019639213.757,
                    1732019638591.941,
                    1732019638019.095,
                    1732019637457.464,
                    1732019636765.253,
                    1732019636030.844,
                    1732019635117.011,
                    1732019634316.524,
                    1732019633482.114,
                    1732019632624.395,
                    1732019631880.414,
                    1732019631229.417,
                    1732019630530.325,
                    1732019629850.227,
                    1732019629098.103,
                    1732019628386.087,
                    1732019627735.068,
                    1732019627129.118,
                    1732019626605.385,
                    1732019626082.227,
                    1732019625445.111,
                    1732019624790.769,
                    1732019623968.533,
                    1732019623183.677,
                    1732019622003.186,
                    1732019620890.175,
                    1732019374912.15,
                    1732019211107.367,
                    1732019150833.26,
                    1732017236396.909,
                    1732016368406.694,
                    1732016064235.315,
                    1732015858563.37,
                    1732015673290.081,
                    1732015613107.176,
                    1732015068355.178,
                    1732013729109.998,
                    1732013275003.28,
                    1732013268869.411,
                    1732012248974.863,
                    1732011565400.281,
                    1732011120880.287,
                    1732010174428.038,
                    1732008516972.52,
                    1732008060006.395,
                    1732007941270.977,
                    1732007412508.855,
                    1732006951024.504,
                    1732006741315.551,
                    1732006353258.046,
                    1732006193244.406,
                    1732006190401.424,
                    1732005592246.415,
                    1732005363101.34,
                    1732004628348.007,
                    1732004463705.603,
                    1732004463323.45,
                    1732004072685.79,
                    1732004069271.539,
                    1732003282267.892,
                    1732002658529.588,
                    1732002503952.811,
                    1732001421246.076,
                    1732001391531.335,
                    1732001144638.565,
                    1732000854811.112
                  ],
                  [
                    315,
                    604,
                    529,
                    754,
                    1336,
                    1534,
                    273,
                    312,
                    258,
                    1323,
                    1443,
                    1457,
                    1461,
                    1504,
                    1305,
                    1346,
                    1396,
                    1622,
                    1361,
                    1458,
                    1410,
                    1377,
                    1571,
                    1618,
                    1318,
                    1362,
                    1626,
                    1296,
                    1712,
                    1525,
                    5689,
                    1434,
                    1283,
                    1591,
                    1390,
                    1396,
                    1389,
                    1402,
                    1342,
                    1351,
                    1362,
                    1326,
                    1401,
                    1355,
                    1619,
                    1435,
                    1835,
                    195,
                    209,
                    288,
                    557,
                    924,
                    343,
                    631,
                    782,
                    1713,
                    1538,
                    581,
                    930,
                    272,
                    689,
                    694,
                    3066,
                    562,
                    576,
                    1538,
                    673,
                    1575,
                    614,
                    581,
                    591,
                    559,
                    512,
                    578,
                    829,
                    331,
                    2089,
                    632,
                    578,
                    817,
                    298,
                    1569,
                    649,
                    729,
                    1488,
                    550,
                    659,
                    220,
                    628
                  ]
                ]
              }
            }
          ]
        }
      ],
      "type": "table",
      "title": "Reproduced with embedded data"
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >table @ 11.2.0</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[jaeger]</td>\n    </tr>\n    \n    <tr><th>Data</th><td> 1 frames, 4 fields, 89 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>Grafana v11.2.0 (2a88694fd3) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"datasource\": {\n    \"default\": false,\n    \"type\": \"jaeger\",\n    \"uid\": \"ee1mwvbydzfuod\"\n  },\n  \"fieldConfig\": {\n    \"defaults\": {\n      \"custom\": {\n        \"align\": \"auto\",\n        \"cellOptions\": {\n          \"type\": \"auto\"\n        },\n        \"inspect\": false\n      },\n      \"mappings\": [],\n      \"thresholds\": {\n        \"mode\": \"absolute\",\n        \"steps\": [\n          {\n            \"color\": \"green\",\n            \"value\": null\n          },\n          {\n            \"color\": \"red\",\n            \"value\": 80\n          }\n        ]\n      },\n      \"color\": {\n        \"mode\": \"thresholds\"\n      }\n    },\n    \"overrides\": []\n  },\n  \"gridPos\": {\n    \"h\": 13,\n    \"w\": 24,\n    \"x\": 0,\n    \"y\": 5\n  },\n  \"id\": 10,\n  \"options\": {\n    \"showHeader\": true,\n    \"cellHeight\": \"sm\",\n    \"footer\": {\n      \"show\": false,\n      \"reducer\": [\n        \"sum\"\n      ],\n      \"countRows\": false,\n      \"fields\": \"\"\n    }\n  },\n  \"pluginVersion\": \"11.2.0\",\n  \"targets\": [\n    {\n      \"datasource\": {\n        \"type\": \"jaeger\",\n        \"uid\": \"ee1mwvbydzfuod\"\n      },\n      \"queryType\": \"search\",\n      \"refId\": \"A\",\n      \"service\": \"ratelimit\",\n      \"tags\": \"\"\n    }\n  ],\n  \"type\": \"table\"\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: undefined // 2024-11-19 10:19:09",
  "tags": [
    "debug",
    "debug-table"
  ],
  "time": {
    "from": "2024-11-19T07:19:09.594Z",
    "to": "2024-11-19T13:19:09.594Z"
  }
}
```
</details>

### Environment (with versions)?

Grafana: 11.2.0
OS: Linux 23.10
Browser: Chrome 131.0.6778.69

### Grafana platform?

Kubernetes

### Datasource(s)?

Jaeger 11.2.0

**Comment by renato-viasoft** [comment]
Great! Thanks again and have a nice day

**Comment by ZaynJiang** [comment]
> Thank you for the reply! I've used the traceId variable on my dashboard and it worked out great! I will check Tempo too! By any chance, does Tempo support multi-tenancy?

I've also encountered this problem. Could you provide an example of implementing dashboard variables? Thank you.

**Comment by renato-viasoft** [comment]
@ZaynJiang, here are some screenshots showing how I configured my dashboard:

1. Dashboard Variable
First, I created this variable:

![Image](https://github.com/user-attachments/assets/177684db-a928-46be-9dcc-696e7548f338)

2. Traces Panel
Then, I used the variable in the query of my traces panel:

![Image](https://github.com/user-attachments/assets/dfec57a7-c9f4-4774-9de8-07de31a034ac)

3. Table Panel
I also created a table panel to list all traces:

![Image](https://github.com/user-attachments/assets/86cfc15e-bc19-45c9-805f-152ce6b7565b)

4. Field Override
Lastly, I added this field override in the table panel to customize the redirect URL for the traceId link:

![Image](https://github.com/user-attachments/assets/48a2f6d3-73f0-4672-8dea-f68a619583b0)

'fduwitnudle68g' is the uid of my dashboard. 
'dashboard-serviços' is the name of my dashboard that contains the panels.
'selected-job' is another variable used to select a specific microservice. It is not relevant for the traces.
